fe1-web/src/features/popcha/screens/PopchaScanner.tsx

```diff
@@ -0,0 +1,199 @@
+   */
+  const verifyScannedInfo = (data: string) => {
+    let url: URL;
+    try {
+      url = new URL(data);
+    } catch (e) {
+      console.log(`Error with scanned url: ${e}`);
+      showErrorMessage('Invalid url');
+      return false;
+    }
+
+    const urlArg = url.searchParams;
+
+    const requiredArguments = [
+      'client_id',
+      'redirect_uri',
+      'login_hint',
+      'nonce',
+      'response_type',
+      'scope',
+    ];
+
+    // Check if all required arguments are present
+    for (const arg of requiredArguments) {
+      if (!urlArg.has(arg)) {
+        showErrorMessage(`Missing argument ${arg}`);
+        return false;
+      }
+    }
+
+    // Check if the response respects openid standard
+    if (urlArg.get('response_type') !== 'id_token token') {
+      showErrorMessage('Invalid response type');
+      return false;
+    }
+
+    if (!(urlArg.get('scope')!.includes('openid') && urlArg.get('scope')!.includes('profile'))) {
+      showErrorMessage('Invalid scope');
+      return false;
+    }
+
+    if (urlArg.has('response_mode')) {
+      if (
+        !(
+          urlArg.get('response_mode')!.includes('query') ||
+          urlArg.get('response_mode')!.includes('fragment')
+        )
+      ) {
+        showErrorMessage('Invalid response mode');
+        return false;
+      }
+    }
+
+    if (urlArg.get('login_hint') !== laoId.toString()) {
+      showErrorMessage('Invalid lao id');
+      console.log(`Scanned lao id: ${urlArg.get('login_hint')}, current lao id: ${laoId}`);
+      return false;
+    }
+
+    return true;
+  };
+
+  /**
+   * Send an auth request to the server
+   * @param data the scanned data
+   * @returns true if the auth request was sent successfully, false otherwise
+   */
+  const sendAuthRequest = async (data: string) => {
+    if (!verifyScannedInfo(data)) {
+      return false;
+    }
+
+    const url = new URL(data);
+    const urlArg = url.searchParams;
+
+    const authResponse = sendPopchaAuthRequest(
+      Hash.fromString(urlArg.get('client_id')!),
+      urlArg.get('nonce')!,
+      urlArg.get('redirect_uri')!,
+      urlArg.get('state'),
+      urlArg.get('response_mode'),
+      laoId,
+      generateToken,
+    );
+
+    return authResponse
+      .then(() => {
+        return true;
+      })
+      .catch((error) => {
+        showErrorMessage(`Could not send auth request: ${error}`);
+        return false;
+      });
+  };
+
+  return (
+    <>
+      <QrCodeScanner
+        showCamera={showScanner}
+        handleScan={(data: string | null) =>
+          data &&
+          sendAuthRequest(data).then((success) => {
+            if (success) {
+              setTextScanned(data);
+              setShowScanner(false);
+            }
+          })
+        }>
+        <View style={styles.container}>
+          <View>
+            <Text style={Typography.paragraph}>Hello, here is your laoID: {laoId}</Text>
+            <text>{textScanned}</text>
```

Comment:
Still needed ? What does it provide ? Or at least having an informal text might be better.

Reply:
Yes `<Text>` is indeed better.